File: backend/cart-service/src/main/java/ecommerce/cartservice/entity/CartItem.java
package ecommerce.cartservice.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "CART_ITEM", schema = "cart_db")
public class CartItem {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "cart_item_seq_gen")
    @SequenceGenerator(name = "cart_item_seq_gen", sequenceName = "SEQ_CART_ITEM_ID", allocationSize = 1)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CART_ID", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Cart cart;

    @Column(name = "PRODUCT_ID", nullable = false)
    private Long productId;

    @Column(name = "QUANTITY", nullable = false)
    private int quantity;

    @Column(name = "UNIT_PRICE")
    private BigDecimal unitPrice;

    @Column(name = "DISCOUNT")
    private BigDecimal discount;

    @Column(name = "IS_SELECTED")
    private Integer isSelected;
}

File: backend/cart/src/entity/CartItem.js
const {EntitySchema} = require('typeorm');

module.exports = new EntitySchema({
    name: 'CartItem',
    tableName: 'CART_ITEM',
    columns: {
        id: {
            type: Number,
            primary: true,
            name: 'ID',
        },
        cartId: {
            type: Number,
            nullable: false,
            name: 'CART_ID',
        },
        productId: {
            type: Number,
            nullable: false,
            name: 'PRODUCT_ID',
        },
        quantity: {
            type: Number,
            name: 'QUANTITY',
        },
        unitPrice: {
            type: 'number',
            precision: 38,
            scale: 2,
            name: 'UNIT_PRICE',
        },
        discount: {
            type: 'number',
            precision: 38,
            scale: 2,
            name: 'DISCOUNT',
        },
    },
    relations: {
        cart: {
            type: 'many-to-one',
            target: 'Cart',
            joinColumn: {
                name: 'CART_ID',
            },
            inverseSide: 'items',
        },
    },
});